From 75a846ea6dce346826376ae2bdd058b5a94fd427 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 6 Mar 2019 15:54:23 +0800
Subject: [PATCH 497/706] gpu: mxc: replace the right header file holding
 phys_to_dma

Give the basic phys_to_dma() and dma_to_phys() helpers a __-prefix and add
the memory encryption mask to the non-prefixed versions.  Use the
__-prefixed versions directly instead of clearing the mask again in
various places.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../kernel/allocator/default/gc_hal_kernel_allocator_dma.c      | 2 +-
 .../kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c   | 2 +-
 .../kernel/allocator/default/gc_hal_kernel_allocator_gfp.c      | 2 +-
 .../allocator/default/gc_hal_kernel_allocator_reserved_mem.c    | 2 +-
 .../kernel/allocator/freescale/gc_hal_kernel_allocator_cma.c    | 2 +-
 drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_allocator.c       | 2 +-
 drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_linux.h           | 2 +-
 drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_os.c              | 2 +-
 8 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dma.c b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dma.c
index bed5f6db0c74..af7408722e88 100644
--- a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dma.c
+++ b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dma.c
@@ -60,7 +60,7 @@
 #include <linux/seq_file.h>
 #include <linux/mman.h>
 #include <asm/atomic.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 #include <linux/slab.h>
 #include <linux/platform_device.h>
 
diff --git a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c
index 5f0e7b44948b..eb57090463a5 100644
--- a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c
+++ b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c
@@ -62,7 +62,7 @@
 #include <linux/slab.h>
 #include <linux/mutex.h>
 #include <asm/atomic.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 
 #include <linux/dma-buf.h>
 #include <linux/platform_device.h>
diff --git a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c
index 937c7521ff0e..19ee8a215af3 100644
--- a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c
+++ b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c
@@ -59,7 +59,7 @@
 #include <linux/seq_file.h>
 #include <linux/mman.h>
 #include <asm/atomic.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 #include <linux/slab.h>
 
 #include "gc_hal_kernel_platform.h"
diff --git a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_reserved_mem.c b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_reserved_mem.c
index a39042fa5462..35dbfdee240e 100644
--- a/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_reserved_mem.c
+++ b/drivers/mxc/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_reserved_mem.c
@@ -59,7 +59,7 @@
 #include <linux/seq_file.h>
 #include <linux/mman.h>
 #include <asm/atomic.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 #include <linux/slab.h>
 #include <linux/io.h>
 #include <linux/ioport.h>
diff --git a/drivers/mxc/hal/os/linux/kernel/allocator/freescale/gc_hal_kernel_allocator_cma.c b/drivers/mxc/hal/os/linux/kernel/allocator/freescale/gc_hal_kernel_allocator_cma.c
index f3f770acb72f..2e32e396b46a 100644
--- a/drivers/mxc/hal/os/linux/kernel/allocator/freescale/gc_hal_kernel_allocator_cma.c
+++ b/drivers/mxc/hal/os/linux/kernel/allocator/freescale/gc_hal_kernel_allocator_cma.c
@@ -60,7 +60,7 @@
 #include <linux/seq_file.h>
 #include <linux/mman.h>
 #include <asm/atomic.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 #include <linux/slab.h>
 #include <linux/platform_device.h>
 
diff --git a/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_allocator.c b/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_allocator.c
index 0eee49090830..1095b0fdc6ba 100644
--- a/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_allocator.c
+++ b/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_allocator.c
@@ -59,7 +59,7 @@
 #include <linux/seq_file.h>
 #include <linux/mman.h>
 #include <asm/atomic.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 #include <linux/slab.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 32)
 #include <linux/anon_inodes.h>
diff --git a/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_linux.h b/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_linux.h
index f8f08d98288c..d28be203c17a 100644
--- a/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_linux.h
+++ b/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_linux.h
@@ -65,7 +65,7 @@
 #include <linux/signal.h>
 #include <linux/interrupt.h>
 #include <linux/vmalloc.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 #include <linux/kthread.h>
 
 #include <linux/idr.h>
diff --git a/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_os.c b/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_os.c
index bc316b3c0acb..fce78a071a1b 100644
--- a/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_os.c
+++ b/drivers/mxc/hal/os/linux/kernel/gc_hal_kernel_os.c
@@ -59,7 +59,7 @@
 #include <linux/seq_file.h>
 #include <linux/mman.h>
 #include <asm/atomic.h>
-#include <linux/dma-mapping.h>
+#include <linux/dma-direct.h>
 #include <linux/slab.h>
 #include <linux/workqueue.h>
 #include <linux/irqflags.h>
-- 
2.17.1

